examples/deploy: manifests improvements

invidian opened the pull request.

rata commented on examples/deploy/podsecuritypolicy.yaml:
```diff
@@ -2,6 +2,39 @@ apiVersion: policy/v1beta1
+    ranges:
+      - min: 1
+        max: 65535
+  readOnlyRootFilesystem: false
```
why false? If we need to write, can't we write to an empty dir and set this to true?

invidian replied:
Nice catch, it should probably be `true`.

invidian replied:
Done.

rata commented on examples/deploy/rbac/cluster-role-binding.yaml:
```diff
@@ -8,5 +8,18 @@ roleRef:
 subjects:
   - kind: ServiceAccount
-    name: flatcar-linux-update-operator-sa
+    name: flatcar-linux-update-operator
```
The tricky thing now is that, if you have it installed, you will have two duplicate service accounts if you just apply the manifests.

Maybe is worth keeping the name while we improve how users should adapt to these changes?

invidian replied:
Perhaps we should have a helm chart for FLUO in the future and we let helm handle that.

I'm not sure how to proceed with it then. Any suggestions?

rata replied:
I think we should not rename and, when we happen to have a more general solution to ship this, we can point to that.

There is little benefit in a rename now. It's more pain than useful.

invidian replied:
Make sense. I'll revert now and let's change it when we add the helm chart, as I think we should do that anyway.

rata replied:
:ok_hand: Please marked as resolved when the revert is pushed :)

invidian replied:
Pushed.

rata commented on examples/deploy/rbac/cluster-role-binding.yaml:
```diff
@@ -8,5 +8,18 @@ roleRef:
+    name: flatcar-linux-update-operator
+    namespace: reboot-coordinator
+---
+kind: ClusterRoleBinding
```
Does it need to be a clusterRoleBinding if we specify the namespace? Why not just a role?

invidian replied:
Can you use `RoleBinding` with `ClusterRole`? Does it then mean, that account in specific namespace has access to given resources in all namespaces? If so, then I think it can be changed to `RoleBinding` :+1: Will need to verify that.

rata replied:
Yes, you can use RoleBinding with ClusterRole. See: https://kubernetes.io/docs/reference/access-authn-authz/rbac/#rolebinding-and-clusterrolebinding

However, it only works for namespaced resources and I see that you want nodes, for example. So, probably need ClusterRoleBinding. Sorry :)

rata commented on examples/deploy/update-agent-sa.yaml:
```diff
@@ -1,5 +1,5 @@
 kind: ServiceAccount
 metadata:
-  name: flatcar-linux-update-operator-sa
+  name: flatcar-linux-update-agent
```
idem

invidian replied:
This one is actually okay, as it changes the service account used and the PR introduces new SA.

rata replied:
I don't see that in the diff... what am I missing?

invidian replied:
Oh, I pushed to the wrong remote :man_facepalming: Updated now.

rata replied:
I don't follow. The diff clearly show you are just modifying the name of this service account. Are you sure you are not mixing it up with another sa?

invidian replied:
Yes. See that `flatcar-linux-update-operator-sa` is now moved to separated file.

rata replied:
Ouch :man_facepalming:

invidian commented:
Thanks for the review @rata. I addressed your comments and tested the manifests again to make sure everything works as expected.

I also added `readOnlyRootFilesystem: true` for both controller and agent and it is working as expected.

rata commented on examples/deploy/update-operator-sa.yaml:
```diff
@@ -0,0 +1,5 @@
+kind: ServiceAccount
+metadata:
+  name: flatcar-linux-update-operator
+  namespace: reboot-coordinator
```
file name and object name don't match

invidian replied:
How they does not match? We have `update-operator-sa.yaml` and `update-agent-sa.yaml`. I don't think they should match.

rata replied:
Oh, haven't noted that. Ok... :)

invidian commented on examples/deploy/podsecuritypolicy.yaml:
```diff
@@ -2,6 +2,39 @@ apiVersion: policy/v1beta1
+    ranges:
+      - min: 1
+        max: 65535
+  readOnlyRootFilesystem: true
```
What do you mean here? That we should modify Lokomotive restricted PSP to have this option enabled as well?

rata replied:
Well, that is out of scope (although not unreasonable, IMHO). But no, I meant to compare both PSPs and make sure no field is missing in the PSP specified here (and compare with the docs too, just in case).

invidian replied:
I'm still not sure what to do here :smile:

rata replied:
Nothing, I just want to double check the diff between this and the restricted PSP in kinvolk/terraform-render-bootkube are similar and check the reference doc to see that no relevant field is missing.

If you want to do something, you can check that the diff is indeed nothing (like `diff -u` output is empty). I can do it later, just don't have much time this week :-/

invidian replied:
I'll double check that then.

rata replied:
Thanks! Seems reasonable! :)

invidian commented:
@rata it seems we use all features as in `restricted` PSP. Here is the list of all PSP fields:
```yaml
apiVersion: policy/v1beta1
kind: PodSecurityPolicy
metadata:
  name: flatcar-linux-update-operator
  annotations:
    seccomp.security.alpha.kubernetes.io/allowedProfileNames: 'docker/default,runtime/default'
    apparmor.security.beta.kubernetes.io/allowedProfileNames: 'runtime/default'
    seccomp.security.alpha.kubernetes.io/defaultProfileName:  'runtime/default'
    apparmor.security.beta.kubernetes.io/defaultProfileName:  'runtime/default'
spec:
  allowPrivilegeEscalation: false
  # allowedCSIDrivers: ?
  # allowedCapabilities: ?
  # allowedFlexVolumes: ?
  # allowedHostPaths: ?
  # allowedProcMountTypes: ?
  # allowedUnsafeSysctls: ?
  # defaultAddCapabilities: ?
  # defaultAllowPrivilegeEscalation: ?
  # forbiddenSysctls: ?
  fsGroup:
    rule: 'MustRunAs'
    ranges:
      - min: 1
        max: 65535
  hostIPC: false
  hostNetwork: false
  hostPID: false
  # hostPorts: ?
  privileged: false
  readOnlyRootFilesystem: true
  requiredDropCapabilities:
    - ALL
  # runAsGroup: ?
  runAsUser:
    rule: 'MustRunAs'
    ranges:
      - min: 1
        max: 65535
  # runtimeClass: ?
  seLinux:
    rule: 'RunAsAny'
  supplementalGroups:
    rule: 'MustRunAs'
    ranges:
      - min: 1
        max: 65535
  volumes:
    - 'secret'
```

The commented out are ones we don't set, except the annotations, which it seems to be, we should add. Any opinions about those?
